fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadScheduler.java

```diff
@@ -45,7 +46,7 @@ protected void runOneCycle() {
         LOG.debug("there are {} job need scheduler", routineLoadJobList.size());
         for (RoutineLoadJob routineLoadJob : routineLoadJobList) {
             // judge nums of tasks more then max concurrent tasks of cluster
-            List<RoutineLoadTask> routineLoadTaskList = null;
+            List<RoutineLoadTaskInfo> routineLoadTaskList = null;
             try {
                 routineLoadJob.writeLock();
```

Comment:
use routineLoadJob.writeLock() to protect is weird.
And lock should be used as (lock() is outside the try{}):
lock();
try {

} finally {
    unlock();
}

Reply:
Using `routineLoadJob.writeLock()` means that every process of different `RoutineLoadJob` will not be blocked.